### PR TITLE
Allow target of calli to be lowered to ldc.i4

### LINKED ISSUE #9481: [Arm64] Crossgen System.Private.CoreLib.ni.dll causes test failures
The following tests are failing when crossgen is run to create System.Private.CoreLib.ni.dll.

JIT/Performance/CodeQuality/Linq/Linq/Linq.sh
JIT/Performance/CodeQuality/Math/Functions/Functions/Functions.sh
JIT/Performance/CodeQuality/Serialization/Deserialize/Deserialize.sh
JIT/Performance/CodeQuality/Serialization/Serialize/Serialize.sh
JIT/Regression/CLR-x86-JIT/v2.1/DDB/b188478/b188478/b188478.sh
JIT/Regression/CLR-x86-JIT/v2.2/ddb/ddb188478/DDB188478/DDB188478.sh
Loader/classloader/regressions/dev11_95728/dev11_95728/dev11_95728.sh

The problem exists on my fork using 1/11/2017 baseline.  Tests were performed on Ubuntu 16.04

Looks like the problem is related to generics.

I need to verify the problem still exists on the tip.

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #9481 

@dotnet/jit-contrib Please see comments in #9481

### CODE REVIEW

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by jashook** [comment]
/cc @dotnet/arm64-contrib @dotnet/jit-contrib 

It looks like the arm64 checked run has been broken. Will look into it.

**Comment by sdmaclea** [comment]
@jashook I faced what looks like the same issues today internally (See #10252).  the root cause for me was using tests which were built before the mscorlib facade was removed by a recent commit.

**Comment by BruceForstall** [comment]
@sdmaclea Is this change still required?

Why does this problem only appear on arm64 Linux, and not arm64 Windows or amd64 Windows?

@briansull PTAL

**Comment by sdmaclea** [comment]
@BruceForstall  Yes, I believe it is necessary.I think it may actually be platform and memory layout specific.  When the target address of the indirect call happens to fit in 4 bytes, lower changes the constant size.  arm64 linux kernel on my platform seems to hands out virtual addresses which have 32 high bits set to zero.

**Comment by BruceForstall** [comment]
@sdmaclea Do you actually mean the importer, not "lower"?

What's the actual IL getting imported for the case that fails? What's the JIT call stack at the assert?

**Comment by sdmaclea** [comment]
@BruceForstall   The IL is injected by C++ code  to instantiate the generics.  The JIT is called on the IL with pretty conservative settings, but the lower code is still running.  I checked the C++ generated IL to have a type I8 with a value something like 0x7fxxxxxx.  I added a watch point on the type and caught code in lower.cpp changing the node type because the value of the constant node fit in an I4.  The only thing lower did was change the node type.

**Comment by sdmaclea** [comment]
@BruceForstall I did not see your whole question....  I will rerun a failing test to get the backtrace.

**Comment by sdmaclea** [comment]
Here is the backtrace.  The IL is generated in frame 22 and compile is started there too.  The address emitted in prestub.cpp:769 creates an LDC.I8 node with a value <2G.  In frame 0 the node is an LDC.I4 node, because it was lowered during compile.

```
#0  Compiler::impImportIndirectCall (this=0x4d4950, sig=<optimized out>, ilOffset=3221225492) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/importer.cpp:5645
#1  0x0000007fb4a93fb4 in Compiler::impImportCall (this=0x4d4950, opcode=CEE_CALLI, pResolvedToken=0x7fffffba40, pConstrainedResolvedToken=<optimized out>, newobjThis=<optimized out>, prefixFlags=<optimized out>, callInfo=0x7fffffbb20, rawILOffset=<optimized out>) at /ho
me/vmjenkins/workspace/Dotnet/build_and_test/src/jit/importer.cpp:6441
#2  0x0000007fb4aaab9c in Compiler::impImportBlockCode (this=<optimized out>, block=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/importer.cpp:12824
#3  0x0000007fb4ab0e64 in Compiler::impImportBlock(BasicBlock*)::$_1::operator()(Compiler::impImportBlock(BasicBlock*)::FilterVerificationExceptionsParam*) const (this=<optimized out>, pParam=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/imp
orter.cpp:15631
#4  Compiler::impImportBlock (this=0x4d4950, block=0x4d9ce0) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/importer.cpp:15641
#5  0x0000007fb4ab3d24 in Compiler::impImport (this=<optimized out>, method=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/importer.cpp:16715
#6  0x0000007fb4a2872c in Compiler::fgImport (this=0x4d4950) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/flowgraph.cpp:6700
#7  0x0000007fb49fd648 in Compiler::compCompile (this=0x4d4950, methodCodePtr=0x7fffffc410, methodCodeSize=0x7fffffc57c, compileFlags=0x7fffffc420) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/compiler.cpp:4206
#8  0x0000007fb49fff34 in Compiler::compCompileHelper (this=0x4d4950, classPtr=<optimized out>, compHnd=<optimized out>, methodInfo=<optimized out>, methodCodePtr=0x7fffffc410, methodCodeSize=0x7fffffc57c, compileFlags=0x7fffffc420, instVerInfo=<optimized out>) at /home/
vmjenkins/workspace/Dotnet/build_and_test/src/jit/compiler.cpp:5799
#9  0x0000007fb49fee40 in Compiler::compCompile(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*)::$_1::operator()(Compiler::compCompile(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*,
CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*)::__JITParam*) const (this=<optimized out>, __JITpParam=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/compiler.cpp:5151
#10 Compiler::compCompile (this=<optimized out>, methodHnd=<optimized out>, classPtr=<optimized out>, compHnd=<optimized out>, methodInfo=<optimized out>, methodCodePtr=<optimized out>, methodCodeSize=<optimized out>, compileFlags=<optimized out>) at /home/vmjenkins/work
space/Dotnet/build_and_test/src/jit/compiler.cpp:5171
#11 0x0000007fb4a00ca8 in jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_3::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_M
ETHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const::{lambda(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_3::operator()(jitNativeCode(CORINFO_METHOD_S
TRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const::__JITParam*)#1}::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, uns
igned int*, JitFlags*, void*)::$_3::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const::__JITParam*) const (this=<optimized out>, __JITpParam=<optimiz
ed out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/compiler.cpp:6450
#12 jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_3::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, u
nsigned int*, JitFlags*, void*)::__JITParam*) const (this=<optimized out>, __JITpParam=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/compiler.cpp:6474
#13 jitNativeCode (methodHnd=<optimized out>, classPtr=<optimized out>, compHnd=<optimized out>, methodInfo=<optimized out>, methodCodePtr=<optimized out>, methodCodeSize=<optimized out>, compileFlags=<optimized out>, inlineInfoPtr=<optimized out>) at /home/vmjenkins/wor
kspace/Dotnet/build_and_test/src/jit/compiler.cpp:6476
#14 0x0000007fb4a0d404 in CILJit::compileMethod (this=<optimized out>, compHnd=0x7fffffc838, methodInfo=0x7fffffc608, flags=<optimized out>, entryAddress=0x7fffffc580, nativeSizeOfCode=0x7fffffc57c) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/ee_il_dll.cpp
:296
#15 0x0000007fb7299e6c in invokeCompileMethodHelper (jitMgr=0x473de0, comp=0x7fffffc838, info=0x7fffffc608, jitFlags=..., nativeEntry=0x7fffffc580, nativeSizeOfCode=0x7fffffc57c) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/jitinterface.cpp:12006
#16 0x0000007fb729a048 in invokeCompileMethod (jitMgr=0x473de0, comp=0x7fffffc838, info=0x7fffffc608, jitFlags=<error reading variable: Cannot access memory at address 0x8800000>, nativeEntry=0x7fffffc580, nativeSizeOfCode=0x7fffffc57c) at /home/vmjenkins/workspace/Dotne
t/build_and_test/src/vm/jitinterface.cpp:12067
#17 CallCompileMethodWithSEHWrapper(EEJitManager*, CEEInfo*, CORINFO_METHOD_INFO*, CORJIT_FLAGS, unsigned char**, unsigned int*, MethodDesc*)::$_4::operator()(CallCompileMethodWithSEHWrapper(EEJitManager*, CEEInfo*, CORINFO_METHOD_INFO*, CORJIT_FLAGS, unsigned char**, un
signed int*, MethodDesc*)::Param*) const (this=<optimized out>, pParam=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/jitinterface.cpp:12126
#18 CallCompileMethodWithSEHWrapper (jitMgr=0x473de0, comp=0x7fffffc838, info=0x7fffffc608, flags=..., nativeEntry=0x7fffffc580, nativeSizeOfCode=0x7fffffc57c, ftn=0x7f3d598d80) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/jitinterface.cpp:12169
#19 0x0000007fb729b664 in UnsafeJitFunction (ftn=0x7f3d598d80, ILHeader=<optimized out>, flags=..., pSizeOfCode=0x7fffffd5b4) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/jitinterface.cpp:12813
#20 0x0000007fb753056c in MethodDesc::MakeJitWorker (this=0x7f3d598d80, ILHeader=<optimized out>, flags=...) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/prestub.cpp:470
#21 0x0000007fb750d030 in JitILStub (pStubMD=0x7f3d598d80) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/dllimport.cpp:5405
#22 0x0000007fb7531008 in CreateUnboxingILStubForSharedGenericValueTypeMethods (pTargetMD=<optimized out>) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/prestub.cpp:794
#23 0x0000007fb753150c in MakeUnboxingStubWorker (pMD=0x104b5c40) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/prestub.cpp:922
#24 0x0000007fb7532314 in MethodDesc::DoPrestub (this=0x104b5c40, pDispatchingMT=0x7f3d5aa018) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/prestub.cpp:1300
#25 0x0000007fb7531cd8 in PreStubWorker (pTransitionBlock=<optimized out>, pMD=0x104b5c40) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/prestub.cpp:1079
#26 0x0000007fb7489db0 in ThePreStub () from /home/sdmaclea/overlay.checked/libcoreclr.so
```

**Comment by sdmaclea** [comment]
@BruceForstall Here is the backtrace for when the LDC_I8 is lowered to LDC_I4.

```
#0  ILCodeStream::LowerOpcode (instr=ILCodeStream::CEE_LDC_I8, pInstr=0x721cc8) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/stubgen.cpp:291
#1  0x0000007fb73c7f78 in ILStubLinker::FirstPassLink (this=0x7fffffd0e8, pInstrBuffer=0x721c58, numInstr=10, pcbCode=0x7fffffccc0, piCurStack=0x7fffffccbc, puMaxStack=0x7fffffccb8) at /home/vmjenkins/worksp
ace/Dotnet/build_and_test/src/vm/stubgen.cpp:720
#2  0x0000007fb73c84d0 in ILStubLinker::Link (this=0x7fffffd0e8, puMaxStack=0x7fffffcddc) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/stubgen.cpp:808
#3  0x0000007fb7324604 in ILStubCache::CreateAndLinkNewILStubMethodDesc (pAllocator=0x7fb7bb6f90 <g_pSystemDomainMemory+3792>, pMT=0x7f3d2325a8, dwStubFlags=2147483680, pSigModule=0x102c1000, pSig=0x10c83c57
 " ", cbSig=3, pTypeContext=0x7fffffd078, pStubLinker=0x7fffffd0e8) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/ilstubcache.cpp:120
#4  0x0000007fb7332bd8 in CreateUnboxingILStubForSharedGenericValueTypeMethods (pTargetMD=0x104dabf8) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/prestub.cpp:799
```

**Comment by sdmaclea** [comment]
ping @briansull @BruceForstall

**Comment by briansull** [comment]
You said: 

> I added a watch point on the type and caught code in lower.cpp changing the node type because the value of the constant node fit in an I4. The only thing lower did was change the node type.

However your change is in the importer and that runs firest way before lower runs.
It seems like your issue is that someone is creating the IL with a LDC.I4 instead of the expected LDC.I8

**Comment by briansull** [comment]
You called it lower (which to a JIT dev only means lower.cpp)

But the actual place that it got changed is in **stubgen.cpp**, where the runtime (i.e. the VM) is creatiung IL for an IL STUB (which is not part of the JIT)

ILCodeStream::LowerOpcode (instr=ILCodeStream::CEE_LDC_I8, pInstr=0x721cc8) at /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/stubgen.cpp:291

**Comment by briansull** [comment]
Probably the correct fix is to change the IL stub generator not to do this "optimization"

**Comment by sdmaclea** [comment]
@briansull Remove "case CEE_LDC_I8:" from LowerOpcode?

**Comment by sdmaclea** [comment]
@briansull Ok revised commit and title

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by BruceForstall** [comment]
@janvorli Can you take a look? Or suggest someone who knows stubgen.cpp?

**Comment by janvorli** [comment]
@BruceForstall I can see that this change basically removes the optimization of any CEE_LDC_I8 instructions in the generated stubs to lower sized ones. I have two questions:
1. Would the JIT generate code that always works with the constant as 64 bit or would it do similar optimization to what the removed code was doing?
2. Couldn't we instead fix the Compiler::impImportIndirectCall so that it would be able to accept a fptr with TYP_I_INT type too? That seems it would be a fix that's more focused on the actual issue that's function address specific. The change in the stublinker has effect on all constant loads no matter whether they are function addresses or not.

**Comment by briansull** [comment]
I don't know the IL _STUB code that well but the goal is here isn't to remove all ability to use the smaller opcodes, but if we have a 64-bit pointer or 64-bit handle that probably should be emitted as a 
CEE_LDC_I8  and not as a CEE_LDC_I4.

 If we have an expression that wants to multiply by 4 or assign 0 to a 64-bit quantity then that can use any of the smaller instructions. 

Choosing between these two different fixes I would prefer to change the JIT at this point.

**Comment by briansull** [comment]
Jan in the 64-bit JIT a TYP_I_INT  is a 64-bit integer
and for the 32-bit JIT a TYP_I_INT  is a 32-bit integer

**Comment by janvorli** [comment]
@briansull I know, I was referring to the 64 bit case that is causing the trouble. The problem is that the code optimizing the stub IL doesn't know if the CEE_LDC_I8 is loading a function pointer or some other constant. The stub generator always generates all constants with CEE_LDC_I8 for 64 bit targets and CEE_LDC_I4 for 32 bit targets and let's the ILCodeStream::LowerOpcode post-process the generated IL and optimize the constant loading instructions.

**Comment by briansull** [comment]
Also based upon this implementation. It seems likely that we must have already encountered this issue on amd64 Windows.  We must have seen a PInvoke to a native method located in the low address space causing us to see this LDC.I4 case already.

**Comment by briansull** [comment]
@janvorli  I agree that we should not have to make this change that always disallows any LDC.I8 instruction from being shorten to a smaller instruction.  I just don't understand why we haven't seen this issue on amd64 Windows.

**Comment by janvorli** [comment]
@briansull I am also surprised by that. I guess the only reason why we have not encountered it on AMD64 too is that by a chance we never get function addresses in the lowest 2GB of the address space on Windows or 64 bit Unix in debug builds. In release builds, the assert would not fire - the question is whether the rest of the JIT code in that code path is resilient to the constant type and so we could just fix it by weakening the assert or if there are other issues lurking elsewhere.

**Comment by sdmaclea** [comment]
@janvorli @briansull  @BruceForstall I had originally weakened the assert based on the comments here. https://github.com/dotnet/coreclr/blob/master/src/jit/vartype.h#L50 .  This implied to me that this type of weakening had occurred in many places when switching to amd64.

I have been running for a few weeks on arm64 with the weakened assert.  I have not seen any issue.

Let me know if I should push the original patch to weaken the assert.

File: src/vm/stubgen.cpp [hunk 1]
```diff
@@ -269,30 +269,6 @@ ILCodeStream::ILInstrEnum ILCodeStream::LowerOpcode(ILInstrEnum instr, ILStubLin
...
-            {
-                instr = (ILInstrEnum)((UINT_PTR)CEE_LDC_I4_0 + uConst);
-            }
-            else
-            if (FitsInI1(uConst))
-            {
-                instr = CEE_LDC_I4_S;
-            }
-            else
-            if (FitsInI4(uConst))
```
Comment (sdmaclea) [inline src/vm/stubgen.cpp]: I we wanted to keep the assert and most of the optimization, only removing this clause, would correspond to the original assert weakening.

**Comment by BruceForstall** [comment]
Going back, the assert you're seeing is in impImportIndirectCall():
``
    GenTreePtr fptr = impPopStack().val;
    assert(genActualType(fptr->gtType) == TYP_I_IMPL);
```
with the type being TYP_INT?

I'm not sure whether the fix should be to allow TYP_INT without any change, or to insert a cast TYP_INT->TYP_I_IMPL on 64-bit platforms.

I'm no ECMA spec guru, but is it legal for CALLI to be given an I4 for the function pointer? That is, is the IL stub generator giving us illegal IL in this case?

**Comment by sdmaclea** [comment]
@BruceForstall "with the type being TYP_INT?" Correct.

**Comment by sdmaclea** [comment]
@jkotas Who is the ECMA script expert to answer @BruceForstall 's question.

**Comment by sdmaclea** [comment]
```
ECMA-335 III.3.20 calli – indirect method call
```
```
The ftn argument must be a method pointer to a method that can be legitimately called with
the arguments described by callsitedescr (a metadata token for a stand-alone signature). 
Such a pointer can be created using the ldftn  or ldvirtftn  instructions, or could have 
been passed in from native code. 
```
```
Correctness: 
Correct CIL requires that the function pointer contains the address of a method whose 
signature is method-signature compatible-with  that specified by callsitedescr  and that 
the arguments correctly correspond to the types of the destination function’s this  
pointer, if required, and parameters. For the purposes of signature matching, the 
HASTHIS  and EXPLICITTHIS flags are ignored; all other items must be identical 
in the two signatures. Unlike Verified CIL, Correct CIL also allows a native int to be 
passed as a byref(&); in which case following the store the value will be tracked by 
garbage collection. 
```

So my naive read seems to say spec is unclear.  Native int may be construed to be 'C' native int, but on some odd platforms an int size does not match a pointer...

Perhaps we should use EmitLDFTN().  Specifically instances of code like this
```
    // 2.5 Push the target address
    pCode->EmitLDC((TADDR)pTargetMD->GetMultiCallableAddrOfCode(CORINFO_ACCESS_ANY));

    // 2.6 Do the calli
    pCode->EmitCALLI(TOKEN_ILSTUB_TARGET_SIG, msig.NumFixedArgs() + 1, msig.IsReturnTypeVoid() ? 0 : 1);
```

Should become 
```
    // 2.5 Push the target address
    Code->EmitLDFTN((TADDR)pTargetMD->GetMultiCallableAddrOfCode(CORINFO_ACCESS_ANY));

    // 2.6 Do the calli
    pCode->EmitCALLI(TOKEN_ILSTUB_TARGET_SIG, msig.NumFixedArgs() + 1, msig.IsReturnTypeVoid() ? 0 : 1);
```

**Comment by janvorli** [comment]
@sdmaclea EmitLDFTN is a very different thing. It gets a function token and produces an address, it does not accept an address. So it cannot be used here.

**Comment by BruceForstall** [comment]
@sdmaclea your first version of this change was:
```
 assert((genActualType(fptr->gtType) == TYP_I_IMPL) || (genActualType(fptr->gtType) == TYP_INT));
```
and that works?

Perhaps we should go back to that. Maybe you should add a comment to explain why the JIT might see a TYP_INT here.

**Comment by briansull** [comment]
I agree that the original change made to weaken the assert in the Jit is the best fix.

**Comment by sdmaclea** [comment]
@BruceForstall Yes the original patch works.  

I think we are all in agreement that the original patch was best.  I will force push it.

**Comment by sdmaclea** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by BruceForstall** [comment]
@mmitche I've seen OSX testing fail basically every PR lately, mostly with "git fetch" failures, e.g.:

https://ci.dot.net/job/dotnet_coreclr/job/master/job/checked_osx10.12_flow_prtest/1093/consoleText

```
 > git fetch --tags --progress https://github.com/dotnet/coreclr.git +refs/heads/*:refs/remotes/origin/* # timeout=90
ERROR: Error cloning remote repo 'origin'
hudson.plugins.git.GitException: Command "git fetch --tags --progress https://github.com/dotnet/coreclr.git +refs/heads/*:refs/remotes/origin/*" returned status code 128:
...
fatal: unable to open .git/objects/pack/tmp_pack_GqNOqa: No such file or directory
fatal: index-pack failed

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1745)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1489)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:64)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:315)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$2.execute(CliGitAPIImpl.java:512)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:152)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:145)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:332)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at hudson.remoting.Engine$1$1.run(Engine.java:90)
	at java.lang.Thread.run(Thread.java:745)
	at ......remote call to dci-mac-build-054(Native Method)
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1433)
	at hudson.remoting.UserResponse.retrieve(UserRequest.java:253)
	at hudson.remoting.Channel.call(Channel.java:797)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:145)
	at sun.reflect.GeneratedMethodAccessor383.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:131)
	at com.sun.proxy.$Proxy73.execute(Unknown Source)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1042)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1082)
	at hudson.scm.SCM.checkout(SCM.java:495)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1278)
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:604)
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:529)
	at hudson.model.Run.execute(Run.java:1720)
	at com.cloudbees.plugins.flow.FlowRun.run(FlowRun.java:153)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
ERROR: null
```

what's up?

**Comment by mmitche** [comment]
The DTAP lab network is not doing so well right now.  Please see the Vs Engineering outage mail sent earlier this morning.  I have attempted to put some mitigation in place but it's unlikely to help that much.

**Comment by BruceForstall** [comment]
@sdmaclea As you can see, we're having some fundamental arm64 (and OSX) CI testing issues right now, so we're going to be blocked on testing for a while -- hopefully not too long. We're going to wait on merging this until we have the right testing back on-line (even though it seems perfectly good and not risky).

**Comment by sdmaclea** [comment]
@BruceForstall Thanks

**Comment by danmosemsft** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test  (network)

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib Looks like the last arm64 run died due to network timeout.  Can someone retrigger it if/when arm64 CI is stable

**Comment by janvorli** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked Build and Test please

**Comment by sdmaclea** [comment]
@janvorli Looks like you are not on the list of users which CI will let trigger arm64 runs.  See https://github.com/dotnet/coreclr/blob/master/netci.groovy#L1066

**Comment by janvorli** [comment]
@sdmaclea right, I've forgotten that. I'll need to add myself to that list.

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Pull request merged**